### DISCUSSION TITLE
How do you handle using multiple OS installations (RHEL and Fedora) with Cobbler?

### DISCUSSION THREAD

**Discussion by tdcam** [cobbler/cobbler#3482] [General]
I have a Cobbler server for my small lab. I have multiple RHEL hypervisors. We use Fedora 38 and RHEL 8 and 9. We default to UEFI instead of BIOS for new VMs on the hypervisors, because a lot of what we do is prototyping builds for physical hardware which has UEFI and Secure Boot enabled.

The problem is, when I kickstart a RHEL image, it works fine because the shim.efi and grubx64.efi in /var/lib/tftp are the ones I copied over from the RHEL RPM. But if I try to kickstart a Fedora VM, it fails when it tries to boot the kernel. I created a shell script that renames shim.efi.fedora grubx64.efi.fedrora to shim.efi and grubx64.efi for when I want to kickstart a Fedora machine, and then it renames shim.efi.rhel and grubx64.efi.rhel to shim.efi and grubx64.efi when I want to kickstart a RHEL VM. But this is clunky as heck.

Does anyone have a smarter way to do this?

**Comment by SchoolGuy** [comment]
The correct way to do this is to build your own shim with your own certificate and roll it out on the hardware (or lab VMs). This is a hell lot more complicated and I don't know if anyone bothered to actually try this end-to-end.

Since the OS vendors use their own shim for each OS, of course, it can't work out of the box easily as we don't know what distro is being booted before the user or automation has chosen the correct shim.

**Reply by tdcam** [reply]
So the way I'm doing it now, just using the right shim for the OS being installed, is probably the easiest way, right? It's a pain, but it seems like it's easier than trying to figure out how to create a custom shim. And since I'm doing this mostly in virtualization, I don't even know how I would create a UEFI image which already has my custom shim installed.

**Reply by SchoolGuy** [reply]
Well in regard to the custom `shim` you would need to host a custom internal package repo that then is utilized by your kickstart to install shim. The trick here would be to use a higher priority than the default repositories to not take the shim from the OS vendor. So yeah to do this properly end-to-end is indeed a hell of an effort.
